### PR TITLE
Refactor/add child types for popover

### LINKED ISSUE #5844: Children typing for PopoverTrigger which will support @types/react 18.0.0 along with backwards compatibility
### Description

The PopoverTrigger does not provide typing to support the upcoming react 18.0.0 types. This issue serves to provide support for @types/react 18.0.0 in addition to backward compatibility. 

If the below solution is suitable, I'll link a PR to it. Please provide me feedback on the proposed solution below, specifically on other props if needed, or let me know if it is too early for this, and I can close it out.

### Problem Statement/Justification

With @types/react 18, implicit children are removed. The current PopoverTrigger has implicit children, and although they are convenient, they can lead to hard-to-detect bugs. Furthermore, this functionality will break in typescript projects with React 18. 

Error with using PopoverTrigger with @types/react 18:
```
Type '{ children: any; }' has no properties in common with type 'IntrinsicAttributes'.
```

### Proposed Solution or API

Provide single react element, string, or number type for PopoverTrigger children:

```ts
interface PopoverTriggerProps {
  /**
   * The React child to use as the
   * trigger for the popover
   */
  children: React.ReactChild
}

export const PopoverTrigger: React.FC<React.PropsWithChildren<PopoverTriggerProps>> = (props) => {
  // enforce a single child
  const child: any = React.Children.only(props.children)
  const { getTriggerProps } = usePopoverContext()
  return React.cloneElement(child, getTriggerProps(child.props, child.ref))
}
```

### Alternatives

_No response_

### Additional Information

- [React 18 removal of implicit children](https://solverfox.dev/writing/no-implicit-children/)
- [React 18 type changes](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56210)

### PR DESCRIPTION
**Pull request by pomSense**
Closes #5844 

## 📝 Description

The PopoverTrigger component breaks with React 18.

## ⛳️ Current behavior (updates)

The child for PopoverTrigger isn't typed, which leads to the following error with react 18:
```
Type '{ children: any; }' has no properties in common with type 'IntrinsicAttributes'.
```

## 🚀 New behavior

Added type to the PopoverTrigger type and specified it accordingly to the PopoverTrigger component:
```ts
interface PopoverTriggerProps {
  /**
   * The React child to use as the
   * trigger for the popover
   */
  children: React.ReactChild
}

export const PopoverTrigger: React.FC<React.PropsWithChildren<PopoverTriggerProps>> = (props) => {
  // enforce a single child
  const child: any = React.Children.only(props.children)
  const { getTriggerProps } = usePopoverContext()
  return React.cloneElement(child, getTriggerProps(child.props, child.ref))
}
```

## 💣 Is this a breaking change (Yes/No):

No.

## 📝 Additional Information
See Issue #5844

### CODE REVIEW

File: packages/popover/src/popover.tsx [hunk 1]
```diff
@@ -76,11 +76,21 @@ if (__DEV__) {
...
+}
+
 /**
  * PopoverTrigger opens the popover's content. It must be an interactive element
  * such as `button` or `a`.
  */
-export const PopoverTrigger: React.FC = (props) => {
+export const PopoverTrigger: React.FC<
+  React.PropsWithChildren<PopoverTriggerProps>
+> = (props) => {
```
Comment (TimKolberger) [inline packages/popover/src/popover.tsx:93]: I think we can simplify this props type definition to

```suggestion
export const PopoverTrigger = (props: PopoverTriggerProps) => {
```

**Review by TimKolberger — CHANGES_REQUESTED** [review]
Thank you for this PR @pomSense 💖 
I added a tiny suggestion.

File: packages/popover/src/popover.tsx [hunk 2]
```diff
@@ -76,11 +76,21 @@ if (__DEV__) {
   PopoverAnchor.displayName = "PopoverAnchor"
 }
 
+export interface PopoverTriggerProps {
+  /**
+   * The React child to use as the
+   * trigger for the popover
+   */
+  children: React.ReactChild
```
Comment (stefee) [inline packages/popover/src/popover.tsx:84]: Note: you might want to use ReactNode if you want to allow all valid children types such as fragments:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/274907dfb7bc98f63fe348defdb0c62774d4483b/types/react/index.d.ts#L221-L229

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/274907dfb7bc98f63fe348defdb0c62774d4483b/types/react/index.d.ts#L784

File: packages/popover/src/popover.tsx (hunk 1 above)
Comment (pomSense) [reply packages/popover/src/popover.tsx]: Thanks, @TimKolberger. Much simpler. Resolved.

File: packages/popover/src/popover.tsx (hunk 2 above)
Comment (pomSense) [reply packages/popover/src/popover.tsx]: Thanks, @stefee, change is made.

**Comment by TimKolberger** [comment]
Thank you @pomSense for working on that! I hope you excuse us for merging #5856 in favor of this PR - we fixed a whole lot of other type issues in the codebase.

@all-contributors please add @pomSense for code.

**Comment by pomSense** [comment]
> Thank you @pomSense for working on that! I hope you excuse us for merging #5856 in favor of this PR - we fixed a whole lot of other type issues in the codebase.
> 
> @all-contributors please add @pomSense for code.

Oh, no problem at all. Thanks for resolving this with #5856 . I'll test out the `next` branch in our codebase to see if we get any other types of issues and I can mention them somewhere if they are not yet handled. 

Appreciate the help @TimKolberger

**Pull request closed without merging**